lib/call-cabal-project-to-nix.nix

```diff
@@ -524,7 +524,11 @@ let
       # some packages that will be excluded by `index-state-found`
       # which is used by cabal (cached-index-state >= index-state-found).
       dotCabal {
-        inherit cabal-install nix-tools extra-hackage-tarballs;
+        inherit cabal-install nix-tools;
+        extra-hackage-tarballs =
+          if __length extra-hackage-tarballs != 0
```

Comment:
Hmmmmmm. I'm not sure about this. I'd have expected maybe:
- Unconditionally adding the extra tarballs
- Having an assertion and failing if you have both repositories from the project and extra tarballs
But just silently dropping the repository ones seems odd and like it might result in mysterious bugs.

Reply:
I have changed `extra-hackage-tarballs` to an attr set that is combined with
```
fixedProject.tarballs // extra-hackage-tarballs
```